transformer_engine/pytorch/attention.py

```diff
@@ -1035,46 +1100,69 @@ def _rotate_half(x: torch.Tensor) -> torch.Tensor:
 
 
 def apply_rotary_pos_emb(
-        t: torch.Tensor,
-        freqs: torch.Tensor,
-        tensor_format: str = "sbhd"
-    ) -> torch.Tensor:
+    t: torch.Tensor,
+    freqs: torch.Tensor,
+    tensor_format: str = "sbhd",
+    fused: bool = False,
+    cu_seqlens: Union[torch.Tensor, None] = None,
+) -> torch.Tensor:
     """
-        Parameters
-        ----------
-        t: torch.Tensor
-            input tensor on which rotary positional embedding will be applied
-        freqs: torch.Tensor
-            rotary positional embeding tensor `freqs` is of shape
-            `[seq_length, ..., dim]`
-        tensor_format: {'sbhd', 'bshd'}, default = 'sbhd'
-            is `bshd` if `t` is of shape `[bs, seq, ...]`, or `sbhd` if `t` is
-            of shape `[seq, bs, ...]`.
+    Apply rotary positional embedding tensor to the input tensor.
 
+    Parameters
+    ----------
+    t: torch.Tensor
+        Input tensor of shape `[s, b, h, d]`, `[s, b, h, d]` or `[t, h, d]`, on which
+        rotary positional embedding will be applied.
+    freqs: torch.Tensor
+        Rotary positional embedding tensor of shape `[s2, 1, 1, d2]` and dtype 'float',
+        with `s2 >= s` and `d2 <= d`.
+    fused: bool, default = False
+        Whether to use a fused applying RoPE implementation.
+        Note: When using fused RoPE on an input tensor in `sbhd` format, the output
+        tensor will be in shape `[s, b, h, d]` but the 's' and 'b' dimensions of the
+        underlying memory layout are transposed.
+        See https://github.com/NVIDIA/TransformerEngine/pull/517 for more details.
```

Comment:
Why is the output not contiguous in the SBHD case? Your kernel should be flexible enough to handle contiguous outputs for both SBHD and BSHD.